src/sentry/api/endpoints/event_file_committers.py

```diff
@@ -157,7 +149,17 @@ def get(self, _, project, event_id):
 
         group = Group.objects.get(id=event.group_id)
 
-        commits = self._get_commits(event.project, group.get_first_release())
+        first_release_version = group.get_first_release()
+
+        if not first_release_version:
+            return Response({'detail': 'Release not found'}, status=404)
+
+        releases = Release.get_closest_releases(project, first_release_version)
```

Comment:
So this is something that you probably didn't intend. What's being returned here is a `QuerySet` and is still not evaluated.

Below at L159, you're evaluating because of the `if not releases` check. Then you keep passing through this `QuerySet` down into `_get_commits` as well which will turn into a nested subquery. While technically fine, you're just evaluating this needlessly.

Maybe do a `releases.exists()` instead of `if not releases` to be a much more efficient query there, then pass along the `QuerySet` to `_get_commits` to do it's thing there.

Or alternatively coerce to a list to evaluate first.

Either of these will be a bit weird because the return value from `get_closest_releases` is either a list or a `QuerySet`. So some refactoring here to make this all work out will be needed.

Reply:
good point. so one solution is just to get rid of the `if not releases` check and just pass the empty list to `self._get_commits`. however, i'm realizing from travis that mysql doesn't support limits in subqueries, so I could just coerce it to a list since that might be necessary anyway.